src/ui/units/dash/store/actions/dashboard/actions.ts

```diff
@@ -0,0 +1,527 @@
+        const {
+            dash: {tabId: stateTabId},
+        } = getState();
+
+        const tabId = selectedTabId || stateTabId;
+
+        const payload: Partial<DashState> = {
+            mode: Mode.SelectState,
+        };
+
+        const search = [];
+
+        if (tabId) {
+            payload.tabId = tabId;
+            search.push(`tab=${tabId}`);
+        }
+
+        if (stateHash) {
+            try {
+                dispatch({type: SET_STATE, payload: {mode: Mode.Loading}});
+
+                const {
+                    dash: {
+                        entry: {entryId},
+                    },
+                } = getState();
+
+                const hashData = await getSdk()
+                    .us.getDashState({entryId, hash: stateHash})
+                    .catch((error) => {
+                        logger.logError('getDashState failed', error);
+                        console.error('STATE_LOAD', error);
+                    });
+
+                if (hashData && tabId) {
+                    // TODO find out from what controls field is
+                    const {controls, ...states} = hashData.data as any;
+                    payload.hashStates = {
+                        [tabId]: {
+                            hash: stateHash,
+                            state: {...controls, ...states},
+                        },
+                    };
+                    search.push(`state=${stateHash}`);
+                }
+            } catch (error) {
+                logger.logError('dash: setSelectStateMode failed', error);
+                console.error('MAILING_MODE_STATE', error);
+            }
+        }
+
+        history.push({...location, search: `?${search.join('&')}`, hash: ''});
+        dispatch({type: SET_STATE, payload});
+    };
+};
+
+export const setEditMode = (successCallback = () => {}, failCallback = () => {}) => {
+    return async function (dispatch: DashDispatch, getState: () => DatalensGlobalState) {
+        const {
+            dash: {
+                entry: {entryId, savedId: stateSavedId, fake},
+                mode,
+            },
+        } = getState();
+
+        if (fake) {
+            return;
+        }
+
+        try {
+            const {savedId} = await getSdk().us.getEntryMeta({entryId});
+
+            if (stateSavedId !== savedId) {
+                dispatch(
+                    openDialogConfirm({
+                        onApply: () => {
+                            location.reload();
+                        },
+                        message: i18n('label_obsolete-version'),
+                        isWarningConfirm: true,
+                        cancelButtonView: 'flat',
+                        confirmButtonView: 'normal',
+                        showIcon: false,
+                        onCancel: () => {
+                            failCallback();
+                            (dispatch as ConnectionsReduxDispatch)(closeDialogConfirm());
```

Comment:
It's strange to use type from connections unit in dash. Maybe it's worth to add `DialogAction` to `DashDispatch` type? 
There are more similar cases below

Reply:
will be fixed later